src/main/java/lotto/domain/LottoTicket.java

```diff
@@ -0,0 +1,40 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class LottoTicket {
+
+    public static final int LOTTO_SIZE = 6;
+
+    private final List<Integer> lottoNumbers;
+
+    public LottoTicket(NumberCreationStrategy strategy) {
+        this(strategy.createNumber());
+    }
+
+    public LottoTicket(List<Integer> lottoNumbers) {
+        validateLottoNumberLength(lottoNumbers);
+        this.lottoNumbers = lottoNumbers;
+    }
+
+    private void validateLottoNumberLength(List<Integer> lottoNumbers) {
+        if (lottoNumbers.size() != LOTTO_SIZE) {
+            throw new IllegalArgumentException("로또 번호는 6자리입니다.");
+        }
+    }
+
+    public int calculateSameNumberCount(LottoTicket lottoTicket) {
+        List<Integer> lottoNumber = lottoTicket.getLottoNumber();
+        return (int) this.lottoNumbers.stream()
+                .filter(lottoNumber::contains)
+                .count();
+    }
+
+    public boolean hasBonusNumber(int number) {
```

Comment:
보너스 번호뿐만 아니라 당첨 번호의 일치 여부도 확인할 수 있으니, 
메서드 이름을 조금 더 범용적으로 변경해도 될 것 같네요.
```suggestion
    public boolean hasNumber(int number) {
```